src/stirling/obj_tools/elf_reader.h

```diff
@@ -50,8 +52,14 @@ class ElfReader {
    * @return error if could not setup elf reader.
    */
   static StatusOr<std::unique_ptr<ElfReader>> Create(
-      const std::string& binary_path,
-      const std::filesystem::path& debug_file_dir = "/usr/lib/debug");
+      const std::string& binary_path, const std::filesystem::path& debug_file_dir = kDebugFileDir);
+
+  /**
+   * Creates an ElfReader that does not enforce the max file size limit. This is useful for cases
+   * where the binary size is known in advance or the binary must be loaded regardless of size.
+   */
+  static StatusOr<std::unique_ptr<ElfReader>> CreateUncapped(
+      const std::string& binary_path, const std::filesystem::path& debug_file_dir = kDebugFileDir);
```

Comment:
From an API perspective, it might be simpler to have a single Create() where the size is an optional parameter that defaults to FLAGS_elf_reader_max_file_size. What do you think?

Reply:
I'm open to changing this, but I wanted to make the uncapped behavior an explicit opt-in for consumers of ElfReader. Since it's used in many places, I could see a future bug where the max file size check is unintentionally bypassed when most consumers should retain the restriction.

Reply:
We talked about this offline, but the suggestion is that the optional parameter defaults to the capped version, so I don't think it exposes any risk. The goal is just to keep the API simpler. But this is a bit subjective, so doesn't need to be a blocker if you prefer the two separate API calls.